components/dataflow/predict/component.yaml

```diff
@@ -0,0 +1,30 @@
+name: Predict using TF on Dataflow
+description: |
+  Runs TensorFlow prediction on Google Cloud Dataflow
+  Input and output data is in GCS
+inputs:
+  - {name: Data file pattern,   type: {GcsUriPattern: [text, CSV]},      description: 'GCS or local path of test file patterns.'}
+  - {name: Schema,              type: {GcsUri: [text, json]},     description: 'GCS json schema file path.'}
+  - {name: Target,              type: String,                     description: 'Name of the column for prediction target.'}
+  - {name: Model,               type: {GcsUri: [Directory, Exported TensorFlow models]}, description: 'GCS or local path of model trained with tft preprocessed data.'}   # Models trained with estimator are exported to base/export/export/123456781 directory.  # Our trainer export only one model. #TODO: Output single model from trainer
+#  - {name: Batch size,          type: Integer,   default: 32,     description: 'Batch size used in prediction.'}
+  - {name: Run mode,            type: {Enum: [local, cloud]},     description: 'Whether to run the job locally or in Cloud Dataflow.'}
+  - {name: GCP project,         type: GCP project,                description: 'The GCP project to run the dataflow job.'}
+  - {name: Predictions dir,     type: {GcsUri: Directory},        description: 'GCS or local directory.'} #Will contain prediction_results-* and schema.json files; TODO: Split outputs and replace dir with single file
+outputs:
+  - {name: Predictions dir,     type: {GcsUri: Directory},        description: 'GCS or local directory.'} #Will contain prediction_results-* and schema.json files; TODO: Split outputs and replace dir with single file
+implementation:
+  container:
+    image: gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:85c6413a2e13da4b8f198aeac1abc2f3a74fe789
```

Comment:
You may need to change how we run the sample test (https://github.com/kubeflow/pipelines/blob/master/test/sample-test/run_test.sh). Instead of replacing images in py file you'll also need to do the same for this file.

Reply:
This is a valid concern. I've worked on tackling this problem and we made a big step forward when sample test stopped rebuilding the images every time. The currently active sample test does not replace the images.

The idea is for the samples to load the components using a specific digest (similar to how we currently use container image tags).
If you update the component tag/digest in the sample, the sample test will run and verify that the sample still works.

We still have work to do to automate and simplify the component code update process. I have a simple `update_component.sh` script which is great for local development, but in out repo we should probably use CloudBuild.

Ideally, when you create a PR where you change component code, the CloudBuild will build the image, create a new component.yaml version and give the PR author the digest or URL of that version to be used in samples.